flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/CalcITCase.scala

```diff
@@ -366,6 +366,59 @@ class CalcITCase extends StreamingTestBase {
     sink.getAppendResults.foreach(result => assertEquals(expected, result))
   }
 
+  @Test
+  def testIfFunctionWithNestedInput(): Unit = {
+    val data = List(
+      Row.of(Row.of("Hello", "Worlds"), Int.box(1)),
+      Row.of(Row.of("Hello", "Hidden"), Int.box(5)),
+      Row.of(Row.of("Hello again", null), Int.box(2)),
+      Row.of(Row.of(null, "World"), Int.box(0)),
+      Row.of(Row.of("Hello again", "Hide"), Int.box(6))
+    )
+    val dataId = TestValuesTableFactory.registerData(data)
+    val ddl =
```

Comment:
Following the style of this test, use the way like
`   val t = env
      .fromCollection(TestData.tupleData3)
      .toTable(tEnv, 'a, 'b, 'c)
    tEnv.createTemporaryView("MyTable", t)
` 
to create a table ? Then the code will look much clear.